### DISCUSSION TITLE
Nginx-Cloudflare如何绑定已启用的端口

### DISCUSSION THREAD

**Discussion by despire1119** [MikeWang000000/Natter#70] [Q&A] [answered]
大佬好，

我想要将我的Cloudflare域名直接绑定到家中NAS主页所在的端口，请问应该如何实现？（yml文件中配置的端口必须是未启用的）
或者是否有办法通过新启用的这个端口去访问其他端口的服务？

谢谢。

**Comment by MikeWang000000** [comment]
我没有太明白你想要达到想效果。
是想要不带端口号，也不跳转，使用域名直接访问？

那这样是做不到的，因为打洞出来的必定不是 `80` 或者 `443` 端口。
有一种选择是 Cloudflare Tunnel，不过那是经过 CDN 的情况，不是直连。

**Reply by despire1119** [reply]
不是的，可以接受跳转，目前示例是跳转到18888，我现在想跳转到2333（NAS系统的操作界面所在的端口），但是如果在nginx中配置port为2333的话，就会抛出“端口已被占用”，是不是我对nginx有什么误解？

**Reply by MikeWang000000** [reply, marked as answer]
是的，确实存在误解。
Nginx 是为了开一个新的 Web 服务器。而 NAS 已经有 Web 服务了，您只需要把 `command: -e /opt/cf-redir.py -p 18888` 中的 `18888` 改成 NAS 的 Web 端口。下面的 `nginx:` 部分可以直接去掉，用不到 Nginx。

**Reply by despire1119** [reply]
自罚一杯
